code/modules/client/preferences_toggles.dm

```diff
@@ -1,162 +1,249 @@
+//this works as is to create a single checked item, but has no back end code for toggleing the check yet
+#define TOGGLE_CHECKBOX(PARENT, CHILD) PARENT/CHILD/abstract = TRUE;PARENT/CHILD/checkbox = CHECKBOX_TOGGLE;PARENT/CHILD/verb/CHILD
+
+//Example usage TOGGLE_CHECKBOX(datum/menu/Settings/Ghost/chatterbox, toggle_ghost_ears)()
+
+//override because we don't want to save preferences twice.
+/datum/menu/Settings/Set_checked(client/C, verbpath)
+	if (checkbox == CHECKBOX_GROUP)
+		C.prefs.menuoptions[type] = verbpath
+	else if (checkbox == CHECKBOX_TOGGLE)
+		var/checked = Get_checked(C)
+		C.prefs.menuoptions[type] = !checked
+		winset(C, "[verbpath]", "is-checked = [!checked]")
+
+/datum/menu/Settings/verb/setup_character()
+	set name = "Game Preferences"
+	set category = "Preferences"
+	set desc = "Open Game Preferences Window"
+	usr.client.prefs.current_tab = 1
+	usr.client.prefs.ShowChoices(usr)
+
 //toggles
-/client/verb/toggle_ghost_ears()
+/datum/menu/Settings/Ghost/chatterbox
+	name = "Chat Box Spam"
+
+TOGGLE_CHECKBOX(/datum/menu/Settings/Ghost/chatterbox, toggle_ghost_ears)()

```

Comment:
Not really, he chose for it to be done this way.
```
#define TOGGLE_CHECKBOX(PARENT, CHILD) PARENT/CHILD/abstract = TRUE;PARENT/CHILD/checkbox = CHECKBOX_TOGGLE;PARENT/CHILD/verb/CHILD
```
Instead of:
```
#define TOGGLE_CHECKBOX(PARENT, CHILD) ##PARENT/##CHILD/abstract = TRUE;##PARENT/##CHILD/checkbox = CHECKBOX_TOGGLE;##PARENT/##CHILD/verb/##CHILD()
```